tiledb/sm/query/reader.cc

```diff
@@ -1391,6 +1413,15 @@ Status Reader::copy_partitioned_fixed_cells(
     const auto& cs = (*result_cell_slabs)[cs_idx];
     uint64_t offset = cs_offsets[cs_idx];
 
+    if (query_datatypes_.find(*name) != query_datatypes_.end()) {
+      double query_size_ratio = calc_query_size_ratio(*name);
+      cell_size = (uint64_t)(query_size_ratio * cell_size);
+      fill_value_size = (uint64_t)(query_size_ratio * fill_value_size);
```

Comment:
same here.

for the fill value, calculate it as `fill_value_size = fill_value_size / cell_size * <size of converted datatype>`

Reply:
yes